src/shared_modules/content_manager/tests/unit/APIDownloader_test.cpp

```diff
@@ -12,12 +12,150 @@
 #include "APIDownloader_test.hpp"
 #include "APIDownloader.hpp"
 #include "updaterContext.hpp"
+#include "gtest/gtest.h"
+#include <filesystem>
 
 /*
  * @brief Tests the instantiation of the APIDownloader class
  */
-TEST_F(APIDownloaderTest, instantiation)
+TEST_F(APIDownloaderTest, TestInstantiation)
 {
+    m_spAPIDownloader.reset();
     // Check that the APIDownloader class can be instantiated
-    EXPECT_NO_THROW(std::make_shared<APIDownloader>());
+    EXPECT_NO_THROW(m_spAPIDownloader = std::make_shared<APIDownloader>());
+}
+
+/**
+ * @brief Tests handle a valid request with raw data.
+ */
+TEST_F(APIDownloaderTest, TestHandleValidRequestWithRawData)
+{
+    m_spUpdaterContext->spUpdaterBaseContext = m_spUpdaterBaseContext;
+
+    const auto& outputFolder {m_spUpdaterContext->spUpdaterBaseContext->outputFolder};
+    const auto& fileName {
+        m_spUpdaterContext->spUpdaterBaseContext->configData.at("contentFileName").get<std::string>()};
+    const auto contentPath {static_cast<std::string>(m_spUpdaterBaseContext->contentsFolder) + "/" + fileName};
+    const auto downloadPath {static_cast<std::string>(m_spUpdaterBaseContext->downloadsFolder) + "/" + fileName};
+
+    EXPECT_NO_THROW(m_spAPIDownloader->handleRequest(m_spUpdaterContext));
+
+    EXPECT_NO_THROW(m_spAPIDownloader->handleRequest(m_spUpdaterContext));
+
+    EXPECT_EQ(m_spUpdaterContext->data.at("paths").at(0), contentPath);
+
+    const auto stageStatus = m_spUpdaterContext->data.at("stageStatus").at(0);
+
+    EXPECT_EQ(stageStatus.at("stage"), "APIDownloader");
+    EXPECT_EQ(stageStatus.at("status"), "ok");
```

Comment:
Nitpick: What about something like ...
```suggestion
    const auto stageStatus = m_spUpdaterContext->data.at("stageStatus");
    const auto expectedStageStatus = R"(
        [
            {
                "stage": "APIDownloader",
                "status": "ok"
            }
        ]
    )"_json;
    EXPECT_EQ(stageStatus, expectedStageStatus);
```

In this way, we can be sure there is no other undesired data on the `stageStatus` object.

Reply:
Same for the other tests.

Reply:
On this particular test the stage status is equal to
```_json
[{"stage":"APIDownloader","status":"ok"},{"stage":"APIDownloader","status":"ok"}]
```
because of the double execution of `handleRequest`.

Reply:
done